parserator_web/views.py

```diff
@@ -14,11 +15,41 @@ class AddressParse(APIView):
     renderer_classes = [JSONRenderer]
 
     def get(self, request):
-        # TODO: Flesh out this method to parse an address string using the
-        # parse() method and return the parsed components to the frontend.
-        return Response({})
+        """Returns the string the user sent, the parsed address
+        components and the address type if the string is a valid address"""
+
+        # Get the unparsed address from JSON object query
+        input_string = request.query_params['address']
```

Comment:
I think I need clarification on this - perhaps we can spare a bit of time to dig into this when we speak next week?

To answer your question: When I first made this PR, this did not handle cases with an empty address submission. I wasn't too worried because I wanted to catch those cases clientside. (And thanks to your feedback, that's implemented [here](https://github.com/Wizardess/code-challenge/pull/1/commits/aa5d0619a4840c8d89e7839220f14837e7525a49).)

Reply:
My question was more along the lines of, will this line break if the "address" key is not in `request.query_params`? How might you revise it so that case is handled gracefully?

Reply:
(Try going to http://localhost:8000/api/parse/ without the address param to see what happens!)